autogluon/utils/tabular/ml/models/lgb/lgb_model.py

```diff
@@ -144,14 +146,36 @@ def predict_proba(self, X, preprocess=True):
             else:  # Should this ever happen?
                 return y_pred_proba[:, 1]
 
+    def preprocess(self, X, is_train=False):
+        X = super().preprocess(X=X)
+
+        if is_train:
+            requires_internal_map = False
+            for column in X.columns:
+                new_column = re.sub(r'[",:{}[\]]', '', column)
+                if new_column != column:
+                    requires_internal_map = True
+                if requires_internal_map:
+                    break
+            if requires_internal_map:
+                self._internal_feature_map = {feature: i for i, feature in enumerate(list(X.columns))}
```

Comment:
Is renaming features as integers `i` the best strategy if user wants to work with lightGBM themselves post-autogluon fit?  Eg. they might want to use lightGBM's feature-importance functionalities.  So perhaps it's better to use map that changes names as little as possible such as your previous: `column` -> `new_column = re.sub(r'[",:{}[\]]', '', column)` (if name uniqueness gets violated, could additionally paste index `i` to the end of `new_column`).

Reply:
Functionality is purely internal, permutation shuffling feature importance output is still in the original names if you look at how I altered the feature importance function.

I have added parameter use_original_feature_names to get_model_feature_importance (the LightGBM specialized output), which advanced users can leverage if necessary. If they want to use LightGBM's specialized feature importance, that is a niche case that isn't representative of most users (Permutation shuffling is better). Also, the amount of problems where this conversion occurs is rare (I've only seen it for NLP problems).

I'd rather have it clear that the column names were altered rather than subtly changing the names and giving people the wrong idea if they are planning to do something programmatically. By giving them the option to convert back to exact original columns, this enables a more seamless experience between different models.